yt_dlp/extractor/nebula.py

```diff
@@ -3,139 +3,126 @@
+            raise
+        self._api_token = response.get('key')
+        if not self._api_token:
+            raise ExtractorError('Login failed: No token')
 
+    def _call_api(self, *args, **kwargs):
+        if self._token:
+            kwargs.setdefault('headers', {})['Authorization'] = f'Bearer {self._token}'
         try:
-            return inner_call()
+            return self._download_json(*args, **kwargs)
         except ExtractorError as exc:
-            # if 401 or 403, attempt credential re-auth and retry
-            if exc.cause and isinstance(exc.cause, urllib.error.HTTPError) and exc.cause.code in (401, 403):
-                self.to_screen(f'Reauthenticating to Nebula and retrying, because last {auth_type} call resulted in error {exc.cause.code}')
-                self._perform_login()
-                return inner_call()
-            else:
+            if not isinstance(exc.cause, urllib.error.HTTPError) or exc.cause.code not in (401, 403):
                 raise
+            username, password = self._get_login_info()
+            if not username:
+                self.raise_login_required(method='password')
+
+            self.to_screen(f'Reauthenticating to Nebula and retrying, '
+                           f'because last API call resulted in error {exc.cause.code}')
+            self._perform_login(username, password)
+            self._real_initialize()
+            return self._download_json(*args, **kwargs)
 
-    def _fetch_nebula_bearer_token(self):
-        """
-        Get a Bearer token for the Nebula API. This will be required to fetch video meta data.
-        """
-        response = self._call_nebula_api('https://api.watchnebula.com/api/v1/authorization/',
-                                         method='POST',
-                                         note='Authorizing to Nebula')
-        return response['token']
+    def _real_initialize(self):
+        self._token = self._download_json(
+            'https://api.watchnebula.com/api/v1/authorization/', None,
+            headers={'Authorization': f'Token {self._api_token}'} if self._api_token else None,
+            note='Authorizing to Nebula', data=b'')['token']
 
-    def _fetch_video_formats(self, slug):
-        stream_info = self._call_nebula_api(f'https://content.api.nebula.app/video/{slug}/stream/',
-                                            video_id=slug,
-                                            auth_type='bearer',
-                                            note='Fetching video stream info')
-        manifest_url = stream_info['manifest']
-        return self._extract_m3u8_formats_and_subtitles(manifest_url, slug, 'mp4')
+    def _extract_formats(self, slug):
+        stream_info = self._call_api(
+            f'https://content.api.nebula.app/video/{slug}/stream/',
+            slug, note='Fetching video stream info')
+        fmts, subs = self._extract_m3u8_formats_and_subtitles(stream_info['manifest'], slug)
```

Comment:
https://github.com/yt-dlp/yt-dlp/pull/7156#issuecomment-1600301485
```suggestion
        fmts, subs = self._extract_m3u8_formats_and_subtitles(stream_info['manifest'], slug, 'mp4')
```